From a3e7185abdcea52aedec2b10b19776307eb98a64 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 15 Sep 2020 11:19:00 +0200
Subject: [PATCH] Also provide re2::re2 target

---
 cpp/cmake_modules/FindRE2.cmake | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/cpp/cmake_modules/FindRE2.cmake b/cpp/cmake_modules/FindRE2.cmake
index ce83e5a4bce..645a20f7c09 100644
--- a/cpp/cmake_modules/FindRE2.cmake
+++ b/cpp/cmake_modules/FindRE2.cmake
@@ -69,4 +69,11 @@ if(RE2_FOUND)
                           PROPERTIES IMPORTED_LOCATION "${RE2_LIB}"
                                      INTERFACE_INCLUDE_DIRECTORIES "${RE2_INCLUDE_DIR}")
   endif()
+  # Some third-party dependencies (namely gRPC) are on the look-out for a lower-case re2 Target.
+  if(NOT TARGET re2::re2)
+    add_library(re2::re2 UNKNOWN IMPORTED)
+    set_target_properties(re2::re2
+                          PROPERTIES IMPORTED_LOCATION "${RE2_LIB}"
+                                     INTERFACE_INCLUDE_DIRECTORIES "${RE2_INCLUDE_DIR}")
+  endif()
 endif()
