diff --git a/cpp/src/plasma/CMakeLists.txt b/cpp/src/plasma/CMakeLists.txt
index 729fba7e9..9a8022745 100644
--- a/cpp/src/plasma/CMakeLists.txt
+++ b/cpp/src/plasma/CMakeLists.txt
@@ -136,13 +136,13 @@ if(ARROW_RPATH_ORIGIN)
   endif()
   set_target_properties(plasma_store_server
                         PROPERTIES INSTALL_RPATH ${_lib_install_rpath})
-elseif(APPLE)
+elseif(NOT MSVC)
   # With OSX and conda, we need to set the correct RPATH so that dependencies
   # are found. The installed libraries with conda have an RPATH that matches
   # for executables and libraries lying in $ENV{CONDA_PREFIX}/bin or
   # $ENV{CONDA_PREFIX}/lib but our test libraries and executables are not
   # installed there.
-  if(NOT "$ENV{CONDA_PREFIX}" STREQUAL "" AND APPLE)
+  if(NOT "$ENV{CONDA_PREFIX}" STREQUAL "")
     set_target_properties(plasma_store_server
                           PROPERTIES BUILD_WITH_INSTALL_RPATH
                                      TRUE
