diff --git a/python/setup.py b/python/setup.py
index 5029480e0..e5f7dc937 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -226,7 +226,7 @@ class build_ext(_build_ext):
             static_lib_option = ''
 
             cmake_options = [
-                '-DPYTHON_EXECUTABLE=%s' % sys.executable,
+                '-DPython3_EXECUTABLE=%s' % sys.executable,
                 static_lib_option,
             ]
 
