From 4aef0c716f62299334ff39020232b088082c0a9d Mon Sep 17 00:00:00 2001
From: Peter Andreas Entschev <peter@entschev.com>
Date: Fri, 27 Oct 2023 01:54:00 -0700
Subject: [PATCH 3/4] Adaptations for Arrow 12

---
 python/pyarrow/_s3fs.pyx | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/python/pyarrow/_s3fs.pyx b/python/pyarrow/_s3fs.pyx
index 7979286..2ad879d 100644
--- a/python/pyarrow/_s3fs.pyx
+++ b/python/pyarrow/_s3fs.pyx
@@ -261,6 +261,25 @@ cdef class S3FileSystem(FileSystem):
                  load_frequency=900, proxy_options=None,
                  allow_bucket_creation=False, allow_bucket_deletion=False,
                  retry_strategy: S3RetryStrategy = AwsStandardS3RetryStrategy(max_attempts=3)):
+        ensure_s3_initialized()
+
+        self._initialize_s3(access_key=access_key, secret_key=secret_key, session_token=session_token,
+                            anonymous=anonymous, region=region, request_timeout=request_timeout,
+                            connect_timeout=connect_timeout, scheme=scheme, endpoint_override=endpoint_override,
+                            background_writes=background_writes, default_metadata=default_metadata,
+                            role_arn=role_arn, session_name=session_name, external_id=external_id,
+                            load_frequency=load_frequency, proxy_options=proxy_options,
+                            allow_bucket_creation=allow_bucket_creation, allow_bucket_deletion=allow_bucket_deletion,
+                            retry_strategy=retry_strategy)
+
+    def _initialize_s3(self, *, access_key=None, secret_key=None, session_token=None,
+                       bint anonymous=False, region=None, request_timeout=None,
+                       connect_timeout=None, scheme=None, endpoint_override=None,
+                       bint background_writes=True, default_metadata=None,
+                       role_arn=None, session_name=None, external_id=None,
+                       load_frequency=900, proxy_options=None,
+                       allow_bucket_creation=False, allow_bucket_deletion=False,
+                       retry_strategy: S3RetryStrategy = AwsStandardS3RetryStrategy(max_attempts=3)):
         cdef:
             CS3Options options
             shared_ptr[CS3FileSystem] wrapped
